Userland/Libraries/LibWeb/XHR/XMLHttpRequest.cpp

```diff
@@ -117,8 +118,12 @@ DOM::ExceptionOr<JS::Value> XMLHttpRequest::response()
     }
     // 6. Otherwise, if this’s response type is "blob", set this’s response object to a new Blob object representing this’s received bytes with type set to the result of get a final MIME type for this.
     else if (m_response_type == Bindings::XMLHttpRequestResponseType::Blob) {
-        // FIXME: Implement this once we have 'Blob'.
-        return DOM::SimpleException { DOM::SimpleExceptionType::TypeError, "XHR Blob type not implemented" };
+        auto blob_part_or_error = try_make_ref_counted<FileAPI::Blob>(m_received_bytes, get_final_mime_type().type());
+        if (blob_part_or_error.is_error())
+            return DOM::UnknownError::create("Out of memory."sv);
+        auto blob = FileAPI::Blob::create_with_global_object(verify_cast<Bindings::WindowObject>(global_object), Vector<FileAPI::BlobPart> { blob_part_or_error.release_value() });
+        blob->set_wrapper(*Bindings::BlobWrapper::create(global_object, blob));
```

Comment:
This looks :catdogsus: - I think the usual way is calling the generated `wrap()` overload?

https://github.com/SerenityOS/serenity/blob/22a627fc1a694b21325aad32cdbb618fe4080253/Userland/Libraries/LibWeb/WebSockets/WebSocket.cpp#L228-L231